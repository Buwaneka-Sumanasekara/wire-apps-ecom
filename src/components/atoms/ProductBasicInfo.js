import React from "react"
import { View, Text } from "react-native"

export default ProductBasicInfo = ({ name, brandName, price }) => {
    return <View className={"flex flex-row justify-between"}>
        <View className={"w-2/3"}>
            <Text className={"text-slate-600 uppercase"}>{brandName}</Text>
            <Text className={"font-bold text-lg"}>{name}</Text>
        </View>
        <View className={"w-1/3 flex flex-row justify-end"}>
            <Text className={"font-bold text-lg"}>{`${price?.amount} ${price?.currency}`}</Text>
        </View>
    </View>
}