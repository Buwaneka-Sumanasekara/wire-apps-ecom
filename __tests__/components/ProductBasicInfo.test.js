import React from 'react';
import renderer from 'react-test-renderer';
import ProductBasicInfo from '../../src/components/atoms/ProductBasicInfo';
import productJson from '../data/product.json';

describe('<ProductBasicInfo />', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<ProductBasicInfo item={productJson} />).toJSON()
        expect(tree).toMatchSnapshot()
    });
});

