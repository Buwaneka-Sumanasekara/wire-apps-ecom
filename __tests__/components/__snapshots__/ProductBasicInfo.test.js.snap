// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductBasicInfo /> rendered sucessfully 1`] = `
<View
  style={
    [
      {
        "display": "flex",
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "width": "66.666667%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#475569",
          },
          {
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Nike
    </Text>
    <Text
      style={
        [
          {
            "fontWeight": "700",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Nike Air Relentless 4 Mens Running Shoes
    </Text>
  </View>
  <View
    style={
      [
        {
          "width": "33.333333%",
        },
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontWeight": "700",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      45 GBP
    </Text>
  </View>
</View>
`;
